File: backend/src/main/java/com/backend/ord/domain/dto/UserProgressDTO.java
package com.backend.ord.domain.dto;

import com.backend.ord.domain.dto.abstracts.DTOBase;
import lombok.*;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class UserProgressDTO extends DTOBase {
    private Integer pointsObtained;

    private UserDTO user;
    private GameDTO game;
}

File: backend/src/main/kotlin/com/backend/ord/domain/persistence/dto/UserProgressDTO.kt
package com.backend.ord.domain.persistence.dto

import com.backend.ord.domain.persistence.dto.game.GameDTOBase
import java.time.Instant
import java.util.*

class UserProgressDTO(
    val id: UUID = UUID.randomUUID(),

    var pointsObtained: Int = 0,

    var user: UserDTO,
    var game: GameDTOBase<*, *>,

    val createdAt: Instant = Instant.now(),
    var updatedAt: Instant = Instant.now(),
)
